api/v1alpha1/clusterextension_types.go

```diff
@@ -77,31 +82,36 @@ type ClusterExtensionSpec struct {
 
 const (
 	// TODO(user): add more Types, here and into init()
-	TypeInstalled = "Installed"
-	TypeResolved  = "Resolved"
+	TypeInstalled      = "Installed"
+	TypeResolved       = "Resolved"
+	TypeHasValidBundle = "HasValidBundle"
```

Comment:
This came over from rukpak, so we want to keep these statuses.
Also, the more information we can provide to the user/customer about what's happening the better.

Reply:
These came over from rukpak. We can handle them in a follow up if we don't want to expose them. See #880

Reply:
But rukpak was operating on a different CRD. I don't think we should be expanding ClusterExtension API surface. Especially because nothing seems to rely on `TypeHasValidBundle` condition type.

Reply:
I'm all for rethinking what information we are presenting in the status, but I agree with @tmshort that it would be fine to handle as a follow up to limit continued churn on this PR. 

This is the initial implementation taking a lot of inspiration from rukpak, let's create an issue to capture that this needs reconsideration.